### PR TITLE
Registry Entry Attachment: verify entry is registered

### LINKED ISSUE #193: Registry Entry Attributes: Verify registry entry is registered
(this is just so I don't inevitably forget about this _again_ - it's important, and also QFAPI already has mechanisms in place to prevent it from being a Fabric compat problem)

### PR DESCRIPTION
**Pull request by Leo40Git**
Closes #193.
Also does some light Checkstyle fixes.

_This PR was created live on stage!_

### CODE REVIEW

**Review by EnnuiL — APPROVED** [review]

**Review by OroArmor — APPROVED** [review]

**Comment by EnnuiL** [comment]
Testing with a broken version of BetterEnd that broke because of the block of charcoal being registered in the fuel map REA earlier than its own registration, I managed to heap multiples of 9 charcoal; Marking this as `s: not working` for now

File: library/data/registry_entry_attachment/src/main/java/org/quiltmc/qsl/registry/attachment/impl/RegistryEntryAttachmentImpl.java [hunk 1]
```diff
@@ -176,6 +176,10 @@ public Iterator<TagEntry<R, V>> tagEntryIterator() {
 
 	@Override
 	public void put(R entry, V value) {
+		if (this.registry.getId(entry) == null) {
```
Comment (Leo40Git) [inline library/data/registry_entry_attachment/src/main/java/org/quiltmc/qsl/registry/attachment/impl/RegistryEntryAttachmentImpl.java:179]: I just realized something: this check does not work with defaulted registries, such as entity types.
Is there a `Registry.contains` method?

**Comment by EnnuiL** [comment]
...turns out the test wasn't correct for this PR due to it actually being part of a QFAPI issue; A proper test case has been especially built for this PR (and the QFAPI issue) which should return a more concrete result

**Review by EnnuiL — APPROVED** [review]
The test went well! Let's do this

**Pull request merged**